src/components/FilmGallery.vue

```diff
@@ -3,30 +3,68 @@
     <div v-if="!cards || !cards.length" class="no_results_text">
       <span>{{ no_films }}</span>
     </div>
-    <div class="container">
-      <div class="row">
-        <div class="col-md-4" v-for="card in cards" :key="card.title">
-          <film-card class="mb-4" :card="card" />
+    <div v-else>
+      <div class="container">
+        <div class="row">
+          <div class="col-md-4" v-for="card in cards" :key="card.title">
+            <film-card class="mb-4" :card="card" />
+          </div>
         </div>
       </div>
+      <div class="load_more text-center">
+        <load-more-button
+          v-if="!pageIsFullyLoaded"
+          @loadMore="loadMore"
+          :text="loadMoreText"
+        />
+      </div>
     </div>
   </div>
 </template>
 
 <script>
 import "../assets/filmGallery.css";
 import FilmCard from "./FilmCard.vue";
-import { I18N } from "../core/constants";
+import LoadMoreButton from "./LoadMoreButton.vue";
+import { APP_DATA, I18N } from "../core/constants";
+import { STATE_KEYS, GETTERS_KEYS, MUTATIONS_KEYS } from "../core/store";
 export default {
   name: "FilmGallery",
-  components: { FilmCard },
+  components: { FilmCard, LoadMoreButton },
   props: {
     cards: { type: Array }
   },
   data() {
     return {
       no_films: I18N["EN"].NO_FILMS
     };
+  },
+  created() {
+    this.$store.commit(
+      MUTATIONS_KEYS.SET_CARDS_ON_PAGE,
+      APP_DATA.CARDS_PER_PAGE
+    );
+    this.$store.commit(MUTATIONS_KEYS.SET_SEARCH_FILMS, []);
+  },
+  methods: {
+    loadMore: function() {
+      let cardsOnPage =
+        this.$store.state[STATE_KEYS.CARDS_ON_PAGE] + APP_DATA.CARDS_PER_PAGE;
+      cardsOnPage = Math.min(
```

Comment:
Let's re-write this to be a function that expects a list of cards and all the limits and without modifying the original list returns a new list of cards to render.
Mutations of data are usually things to avoid, especially when it's all about the data - which could be huge to process.
That way you're able to transform/filter the data without the data loss + introduce performance optimisations

Reply:
I made bad naming for cardsOnPage. It's not the array of cards, it's just a number that responsible for how many cards should be displayed. I don't change the array of cards. I renamed it to numberOfCardsOnPage